src/server/transaction.cc

```diff
@@ -1066,6 +1069,11 @@ bool Transaction::IsActive(ShardId sid) const {
   return shard_data_[SidToId(sid)].local_mask & ACTIVE;
 }
 
+uint16_t Transaction::GetLocalMask(ShardId sid) const {
+  DCHECK(IsActive(sid));
+  return shard_data_[SidToId(sid)].local_mask;
+}
```

Comment:
One more possible DCHECK is checking the barrier is active, _but_ we print the local mask in test asserts (when the watchdog times out), so asserting a wrong condition would crash error printing 🙄

Reply:
so what do you want to do here? we can also use ` shard_data_[SidToId(sid)].local_mask` when printing

Reply:
we call it in blocking_controller which also crashes